lib/class-wp-rest-post-lock-controller.php

```diff
@@ -0,0 +1,445 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Lock_Controller class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to manage post locks.
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Post_Lock_Controller extends WP_REST_Controller {
+	/**
+	 * Post type.
+	 *
+	 * @var string
+	 */
+	private $post_type;
+
+	/**
+	 * Parent controller.
+	 *
+	 * @var WP_REST_Controller
+	 */
+	private $parent_controller;
+
+	/**
+	 * The base of the parent controller's route.
+	 *
+	 * @var string
+	 */
+	private $parent_base;
+
+	/**
+	 * Whether the controller supports batching.
+	 *
+	 * @var array
+	 */
+	protected $allow_batch = array( 'v1' => true );
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $post_type Post type.
+	 */
+	public function __construct( $post_type ) {
+		$this->post_type         = $post_type;
+		$this->rest_base         = 'lock';
+		$post_type_object        = get_post_type_object( $post_type );
+		$this->parent_base       = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
+		$this->namespace         = ! empty( $post_type_object->rest_namespace ) ? $post_type_object->rest_namespace : 'wp/v2';
+		$this->parent_controller = $post_type_object->get_rest_controller();
+
+		if ( ! $this->parent_controller ) {
+			$this->parent_controller = new WP_REST_Posts_Controller( $post_type );
+		}
+	}
+
+	/**
+	 * Registers the routes for post locking.
+	 *
+	 * @since 4.7.0
```

Comment:
Nit: the version number doesn't seem to be correct. Should we use `12.4.0` or `12.5.0` instead?